server/src/main/java/io/druid/server/coordinator/CoordinatorDynamicConfig.java

```diff
@@ -45,20 +46,60 @@
-      @JsonProperty("replicantLifetime") int replicantLifetime,
-      @JsonProperty("replicationThrottleLimit") int replicationThrottleLimit,
-      @JsonProperty("balancerComputeThreads") int balancerComputeThreads,
-      @JsonProperty("emitBalancingStats") boolean emitBalancingStats,
+      @JacksonInject JacksonConfigManager configManager,
+      @JsonProperty("millisToWaitBeforeDeleting") Long millisToWaitBeforeDeleting,
+      @JsonProperty("mergeBytesLimit") Long mergeBytesLimit,
+      @JsonProperty("mergeSegmentsLimit") Integer mergeSegmentsLimit,
+      @JsonProperty("maxSegmentsToMove") Integer maxSegmentsToMove,
+      @JsonProperty("replicantLifetime") Integer replicantLifetime,
+      @JsonProperty("replicationThrottleLimit") Integer replicationThrottleLimit,
+      @JsonProperty("balancerComputeThreads") Integer balancerComputeThreads,
+      @JsonProperty("emitBalancingStats") Boolean emitBalancingStats,
 
       // Type is Object here so that we can support both string and list as
       // coordinator console can not send array of strings in the update request.
       // See https://github.com/druid-io/druid/issues/3055
       @JsonProperty("killDataSourceWhitelist") Object killDataSourceWhitelist,
-      @JsonProperty("killAllDataSources") boolean killAllDataSources
+      @JsonProperty("killAllDataSources") Boolean killAllDataSources
+  )
+  {
+    CoordinatorDynamicConfig current = configManager.watch(
+        CoordinatorDynamicConfig.CONFIG_KEY,
+        CoordinatorDynamicConfig.class
+    ).get();
+    if (current == null) {
+      current = new Builder().build();
+    }
+
+    this.millisToWaitBeforeDeleting = millisToWaitBeforeDeleting == null ? current.getMillisToWaitBeforeDeleting() : millisToWaitBeforeDeleting;
+    this.mergeBytesLimit = mergeBytesLimit == null ? current.getMergeBytesLimit() : mergeBytesLimit;
+    this.mergeSegmentsLimit = mergeSegmentsLimit == null ? current.getMergeSegmentsLimit() : mergeSegmentsLimit;
+    this.maxSegmentsToMove = maxSegmentsToMove == null ? current.getMaxSegmentsToMove() : maxSegmentsToMove;
+    this.replicantLifetime = replicantLifetime == null ? current.getReplicantLifetime() : replicantLifetime;
+    this.replicationThrottleLimit = replicationThrottleLimit == null ? current.getReplicationThrottleLimit() : replicationThrottleLimit;
+    this.balancerComputeThreads = Math.max(balancerComputeThreads == null ? current.getBalancerComputeThreads() : balancerComputeThreads, 1);
+    this.emitBalancingStats = emitBalancingStats == null ? current.emitBalancingStats() : emitBalancingStats;
+
+
+    this.killAllDataSources = killAllDataSources == null ? current.isKillAllDataSources() : killAllDataSources;
+    this.killDataSourceWhitelist = killDataSourceWhitelist == null ? current.getKillDataSourceWhitelist() : parseKillDataSourceWhitelist(killDataSourceWhitelist);
+
+    if (this.killAllDataSources && !this.killDataSourceWhitelist.isEmpty()) {
```

Comment:
Seems that DruidCoordinatorSegmentKiller doesn't have to check the same condition [here](https://github.com/druid-io/druid/blob/master/server/src/main/java/io/druid/server/coordinator/helper/DruidCoordinatorSegmentKiller.java#L88). Would you remove it?